include/common/arch/mips/asm/utils.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __UTILS_H__
+#define __UTILS_H__
+
+
+/*new file :from kernel/mips
+* fixme: gysun
+*
+*/
+
+#ifdef CONFIG_SMP
+# define kernel_uses_llsc	1
+#else
+# define kernel_uses_llsc	0
+#endif
+
+typedef struct {
+	int counter;
+}atomic_t;
+
+
+#if defined(CONFIG_WEAK_REORDERING_BEYOND_LLSC) && defined(CONFIG_SMP)
+#define __WEAK_LLSC_MB		"	sync	\n"
```

Comment:
It shouldn't be compile-time option - instead, consider it's always true and add
```
/*
 * FIXME: detect with compel_cpu_has_feature() if LL/SC implicitly
 * provide a memory barrier.
 */
```